SLF4J: Class path contains multiple SLF4J bindings.

```sh
SLF4J: Found binding in [jar:file:/home/travis/build/CellProfiler/CellProfiler/imagej/jars/slf4j-simple-1.7.7.jar!/org/slf4j/impl/StaticLoggerBinder.class]

SLF4J: Found binding in [jar:file:/home/travis/build/CellProfiler/CellProfiler/imagej/jars/logback-classic-1.1.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]

SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.

SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
```

I've been getting a version of this running from source for some time now; it wasn't always the case, not sure when it started happening or why:
```
running build_ext
skipping '_cpmorphology2.c' Cython extension (up-to-date)
skipping '_watershed.c' Cython extension (up-to-date)
skipping '_propagate.c' Cython extension (up-to-date)
skipping '_filter.c' Cython extension (up-to-date)
skipping '_lapjv.c' Cython extension (up-to-date)
skipping '_convex_hull.c' Cython extension (up-to-date)
skipping '_fastemd.cpp' Cython extension (up-to-date)
running build_ext
java.lang.IllegalArgumentException: Cannot handle replace call to list in ij.Men
us's public static synchronized java.lang.String[] getPlugins()
        at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:724)
        at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:649)
        at net.imagej.patcher.LegacyExtensions.addExtraPlugins(LegacyExtensions.java:475)
        at net.imagej.patcher.LegacyExtensions.injectHooks(LegacyExtensions.java:175)
        at net.imagej.patcher.LegacyInjector.inject(LegacyInjector.java:308)
        at net.imagej.patcher.LegacyInjector.injectHooks(LegacyInjector.java:109)
        at net.imagej.patcher.LegacyEnvironment.initialize(LegacyEnvironment.java:101)
        at net.imagej.patcher.LegacyEnvironment.applyPatches(LegacyEnvironment.java:495)
        at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:397)
        at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:376)
        at net.imagej.patcher.JavaAgent.preinit(JavaAgent.java:147)
        at net.imagej.patcher.JavaAgent.premain(JavaAgent.java:95)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
        at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.lang.RuntimeException: java.io.IOException: invalid constant type: 15
        at javassist.CtClassType.getClassFile2(CtClassType.java:203)
        at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:110)
        at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:96)
        at javassist.compiler.TypeChecker.atMethodCallCore(TypeChecker.java:704)
        at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:681)
        at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:156)
        at javassist.compiler.ast.CallExpr.accept(CallExpr.java:45)
        at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:241)
        at javassist.compiler.CodeGen.atStmnt(CodeGen.java:329)
        at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
        at javassist.compiler.CodeGen.atIfStmnt(CodeGen.java:390)
        at javassist.compiler.CodeGen.atStmnt(CodeGen.java:354)
        at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
        at javassist.compiler.Javac.compileStmnt(Javac.java:568)
        at javassist.expr.MethodCall.replace(MethodCall.java:234)
        at net.imagej.patcher.CodeHacker$7.edit(CodeHacker.java:685)
        at javassist.expr.ExprEditor.loopBody(ExprEditor.java:191)
        at javassist.expr.ExprEditor.doit(ExprEditor.java:90)
        at javassist.CtBehavior.instrument(CtBehavior.java:618)
        at net.imagej.patcher.CodeHacker$EagerExprEditor.instrument(CodeHacker.java:1278)
        at net.imagej.patcher.CodeHacker.replaceCallInMethod(CodeHacker.java:669)
        ... 17 more
Caused by: java.io.IOException: invalid constant type: 15
        at javassist.bytecode.ConstPool.readOne(ConstPool.java:1027)
        at javassist.bytecode.ConstPool.read(ConstPool.java:970)
        at javassist.bytecode.ConstPool.<init>(ConstPool.java:127)
        at javassist.bytecode.ClassFile.read(ClassFile.java:693)
        at javassist.bytecode.ClassFile.<init>(ClassFile.java:85)
        at javassist.CtClassType.getClassFile2(CtClassType.java:190)
        ... 37 more
java.lang.IllegalArgumentException: Cannot handle app name in ij.ImageJ's public
 <init>(java.applet.Applet applet, int mode)
        at net.imagej.patcher.CodeHacker.replaceAppNameInCall(CodeHacker.java:445)
        at net.imagej.patcher.LegacyExtensions.insertAppNameHooks(LegacyExtensions.java:404)
        at net.imagej.patcher.LegacyExtensions.injectHooks(LegacyExtensions.java:290)
        at net.imagej.patcher.LegacyInjector.inject(LegacyInjector.java:308)
        at net.imagej.patcher.LegacyInjector.injectHooks(LegacyInjector.java:109)
        at net.imagej.patcher.LegacyEnvironment.initialize(LegacyEnvironment.java:101)
        at net.imagej.patcher.LegacyEnvironment.applyPatches(LegacyEnvironment.java:495)
        at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:397)
        at net.imagej.patcher.LegacyInjector.preinit(LegacyInjector.java:376)
        at net.imagej.patcher.JavaAgent.preinit(JavaAgent.java:147)
        at net.imagej.patcher.JavaAgent.premain(JavaAgent.java:95)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
        at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.lang.RuntimeException: java.io.IOException: invalid constant type: 15
        at javassist.CtClassType.getClassFile2(CtClassType.java:203)
        at javassist.CtClassType.subtypeOf(CtClassType.java:303)
        at javassist.CtClassType.subtypeOf(CtClassType.java:318)
        at javassist.compiler.MemberResolver.compareSignature(MemberResolver.java:247)
        at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:119)
        at javassist.compiler.MemberResolver.lookupMethod(MemberResolver.java:96)
        at javassist.compiler.TypeChecker.atMethodCallCore(TypeChecker.java:704)

        at javassist.compiler.TypeChecker.atCallExpr(TypeChecker.java:681)
        at javassist.compiler.JvstTypeChecker.atCallExpr(JvstTypeChecker.java:156)
        at javassist.compiler.ast.CallExpr.accept(CallExpr.java:45)
        at javassist.compiler.CodeGen.doTypeCheck(CodeGen.java:241)
        at javassist.compiler.CodeGen.atStmnt(CodeGen.java:329)
        at javassist.compiler.ast.Stmnt.accept(Stmnt.java:49)
        at javassist.compiler.Javac.compileStmnt(Javac.java:568)
        at javassist.expr.MethodCall.replace(MethodCall.java:234)
        at net.imagej.patcher.CodeHacker$4.edit(CodeHacker.java:426)
        at net.imagej.patcher.CodeHacker$4.edit(CodeHacker.java:440)
        at javassist.expr.ExprEditor.loopBody(ExprEditor.java:219)
        at javassist.expr.ExprEditor.doit(ExprEditor.java:90)
        at javassist.CtBehavior.instrument(CtBehavior.java:618)
        at net.imagej.patcher.CodeHacker$EagerExprEditor.instrument(CodeHacker.java:1278)
        at net.imagej.patcher.CodeHacker.replaceAppNameInCall(CodeHacker.java:401)
        ... 16 more
Caused by: java.io.IOException: invalid constant type: 15
        at javassist.bytecode.ConstPool.readOne(ConstPool.java:1027)
        at javassist.bytecode.ConstPool.read(ConstPool.java:970)
        at javassist.bytecode.ConstPool.<init>(ConstPool.java:127)
        at javassist.bytecode.ClassFile.read(ClassFile.java:693)
        at javassist.bytecode.ClassFile.<init>(ClassFile.java:85)
        at javassist.CtClassType.getClassFile2(CtClassType.java:190)
        ... 37 more
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/C:/Trunk/cellprofiler/imagej/jars/slf4j-simpl
e-1.7.7.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/C:/Trunk/cellprofiler/imagej/jars/logback-cla
ssic-1.1.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
```

--- a/cellprofiler/utilities/cpjvm.py
+++ b/cellprofiler/utilities/cpjvm.py
@@ -141,6 +141,37 @@ def get_jars():
             logger.warning("Failed to find tools.jar")
     return class_path
     
+def find_logback_xml():
+    '''Find the location of the logback.xml file for Java logging config
+    
+    Paths to search are the current directory, the utilities directory
+    and ../../java/src/main/resources
+    '''
+    paths = [os.curdir,
+             os.path.dirname(__file__),
+             os.path.join(
+                 os.path.dirname(os.path.dirname(os.path.dirname(__file__))),
+                 "java", "src", "main", "resources")]
+    for path in paths:
+        target = os.path.join(path, "logback.xml")
+        if os.path.isfile(target):
+            return target
+        
+def add_logback_xml_arg(args):
+    '''Add the logback.xml configuration arg if appropriate
+    
+    args: the args to send to the JVM.
+    '''
+    logback_path = find_logback_xml()
+    if logback_path is not None:
+        if sys.platform.startswith("win"):
+            logback_path = logback_path.replace("\\", "/")
+            if logback_path[1] == ':':
+                # \\localhost\x$ is same as x:
+                logback_path = "//localhost/" + logback_path[0] + "$" + \
+                    logback_path[2:]
+        args.append("-Dlogback.configurationFile=%s" % logback_path)
+                          
 def cp_start_vm():
     '''Start CellProfiler's JVM via Javabridge
     
@@ -151,12 +182,10 @@ def cp_start_vm():
     cpprefs.get_awt_headless() - controls java.awt.headless to prevent
         awt from being invoked
     '''
-    
     args = ["-Dloci.bioformats.loaded=true",
-            "-Dlogback.configurationFile=logback.xml",
             "-Djava.util.prefs.PreferencesFactory="+
             "org.cellprofiler.headlesspreferences.HeadlessPreferencesFactory"]
-
+    add_logback_xml_arg(args)
     class_path = get_jars()
     args += get_patcher_args(class_path)
     awt_headless = cpprefs.get_awt_headless()

--- a/cpnose.py
+++ b/cpnose.py
@@ -29,7 +29,7 @@ from nose.util import resolve_name
 from nose.plugins.manager import PluginManager
 
 from cellprofiler.utilities.cpjvm import \
-     get_path_to_jars, get_jars, get_patcher_args
+     get_path_to_jars, get_jars, get_patcher_args, add_logback_xml_arg
 
 import numpy as np
 np.seterr(all='ignore')
@@ -58,6 +58,7 @@ class CPShutdownPlugin(nose.plugins.Plugin):
                 "-Djava.util.prefs.PreferencesFactory="+
                 "org.cellprofiler.headlesspreferences"+
                 ".HeadlessPreferencesFactory"]
+            add_logback_xml_arg(jvm_args)
             #
             # Find the ij1patcher
             #
@@ -87,6 +88,7 @@ class CPShutdownPlugin(nose.plugins.Plugin):
             self.temp_exampleimages = None
         else:
             self.temp_exampleimages = tempfile.mkdtemp(prefix="cpexampleimages")
+            os.environ["CP_EXAMPLEIMAGES"] = self.temp_exampleimages
 
         if "CP_TEMPIMAGES" in os.environ:
             self.temp_images = None

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -144,6 +144,10 @@
 	 			<artifactId>javassist</artifactId>
 	 			<groupId>javassist</groupId>
 	 		</exclusion>
+	 		<exclusion>
+	 			<artifactId>slf4j-simple</artifactId>
+	 			<groupId>org.slf4j</groupId>
+	 		</exclusion>
 	 	</exclusions>
  	</dependency>
  	<!-- 

